src/Compilers/CSharp/Portable/Compilation/CSharpSemanticModel.cs

```diff
@@ -571,22 +576,22 @@ private static SymbolInfo GetSymbolInfoFromSymbolOrNone(ITypeSymbol type)
         /// Given a variable designation (typically in the left-hand-side of a deconstruction declaration statement),
         /// figure out its type by looking at the declared symbol of the corresponding variable.
         /// </summary>
-        private ITypeSymbol TypeFromVariable(SingleVariableDesignationSyntax variableDesignation, CancellationToken cancellationToken)
+        private (ITypeSymbol Type, CodeAnalysis.NullableAnnotation annotation) TypeFromVariable(SingleVariableDesignationSyntax variableDesignation, CancellationToken cancellationToken)
         {
             var variable = GetDeclaredSymbol(variableDesignation, cancellationToken);
 
             if (variable != null)
             {
-                switch (variable.Kind)
+                switch (variable)
                 {
-                    case SymbolKind.Local:
-                        return ((ILocalSymbol)variable).Type;
-                    case SymbolKind.Field:
-                        return ((IFieldSymbol)variable).Type;
+                    case ILocalSymbol local:
```

Comment:
>case ILocalSymbol local: [](start = 20, length = 24)

I am not sure what is the motivation for switching on a type.

Reply:
For both of these cases, I would have had to store the cast to a local variable so I can get at both the type and annotation, and all of Andy and Neal's benchmarking here has indicated that this method is just as fast as the kind test, while being shorter and removing an extra `if` that becomes unecessary.

---
In reply to: [269792200](https://github.com/dotnet/roslyn/pull/34429#discussion_r269792200) [](ancestors = 269792200)